### PR TITLE
v2 of the new generator for redux

### PR DESCRIPTION
**Pull request by justin808**
rails generate react_on_rails:install --redux

Updated as of:

https://github.com/shakacode/react_on_rails/pull/584/commits/fb483874f2bcac9c12e5b1cabd913c7078714c38
fb48387 Update redux example generator spec

### CODE REVIEW

**Comment by justin808** [comment]
@robwise and @nostophilia I updated the PR. Let's make sure this is PERFECT with regards to how we want our redux generation.

File: client/app/bundles/HelloWorld/store/helloWorldStore.jsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { createStore, applyMiddleware } from 'redux';
+import rootReducer from '../reducers/index';
+import Immutable from 'immutable';
+
+// See
+// https://github.com/gaearon/redux-thunk and http://redux.js.org/docs/advanced/AsyncActions.html
+// This is not actually used for this simple example, but you'd probably want to use this
+// once your app has asynchronous actions.
+import thunkMiddleware from 'redux-thunk';
```
Comment (justin808) [inline client/app/bundles/HelloWorld/store/helloWorldStore.jsx:9]: We might want to remove this, as I'm using redux-saga these days. It's a bit confusing to have extra stuff that's not used.

File: client/app/bundles/HelloWorld/containers/HelloWorld.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import React, { PropTypes } from 'react';
+import HelloWorldWidget from '../components/HelloWorldWidget';
+import { connect } from 'react-redux';
+import Immutable from 'immutable';
```
Comment (justin808) [inline client/app/bundles/HelloWorld/containers/HelloWorld.jsx:4]: we might want to remove immutable.js as well.

**Review by justin808 — COMMENTED** [review]
There are comments not yet addressed from here:
https://github.com/shakacode/react_on_rails-test-new-redux-generation/pull/1

Also, @nostophilia please ensure this passes linting.

**Comment by justin808** [comment]
@nostophilia you have commit access to this repo, so once you make more changes, please make a new PR with them.

File: client/app/bundles/HelloWorld/containers/HelloWorld.jsx (hunk 2 above)
Comment (mongjong59) [reply client/app/bundles/HelloWorld/containers/HelloWorld.jsx:4]: @justin808 
So we remove Immutable.js and have the root state as something like { name: "" }?

File: client/app/bundles/HelloWorld/store/helloWorldStore.jsx (hunk 1 above)
Comment (justin808) [reply client/app/bundles/HelloWorld/store/helloWorldStore.jsx:9]: @nostophilia can you remove this?

File: client/app/bundles/HelloWorld/containers/HelloWorld.jsx (hunk 2 above)
Comment (justin808) [reply client/app/bundles/HelloWorld/containers/HelloWorld.jsx:4]: @nostophilia We could. @robwise and @alexfedoseev Please take a look at the code that's generated here.

**Comment by justin808** [comment]
@nostophilia you can create the linter branch like this:

```
➜  ~/shakacode/react-on-rails/tmp/test-new-redux-generation (run-generator-redux-updated-v2 u=) gco -b v2-with-linting                                                                         [17:16:34]
Switched to a new branch 'v2-with-linting'
➜  ~/shakacode/react-on-rails/tmp/test-new-redux-generation (v2-with-linting) git cherry-pick 6cfec059658a1a66afd13aded7399f94142aa358                                                         [17:16:56]
[v2-with-linting 600814d] add eslint
 Date: Sun Oct 30 15:09:30 2016 -1000
 3 files changed, 36 insertions(+), 1 deletion(-)
 create mode 100644 client/.eslintignore
 create mode 100644 client/.eslintrc.yml
➜  ~/shakacode/react-on-rails/tmp/test-new-redux-generation (v2-with-linting) npm i
```

**Comment by justin808** [comment]
@nostophilia we don't pass linting with the shakacode linter.

```
cd client

➜  ~/shakacode/react-on-rails/tmp/test-new-redux-generation/client (v2-with-linting) npm run eslint                                                                                        1 ↵ [17:21:53]

> react-webpack-rails-tutorial@0.0.1 eslint /Users/justin/shakacode/react-on-rails/tmp/test-new-redux-generation/client
> eslint --ext .js,.jsx .

The react/require-extension rule is deprecated. Please use the import/extensions rule from eslint-plugin-import instead.

/Users/justin/shakacode/react-on-rails/tmp/test-new-redux-generation/client/app/bundles/HelloWorld/actions/helloWorldActionCreators.jsx
  3:1  error  Prefer default export  import/prefer-default-export

/Users/justin/shakacode/react-on-rails/tmp/test-new-redux-generation/client/app/bundles/HelloWorld/components/HelloWorldWidget.jsx
  30:11  error  Form controls using a label to identify them must be programmatically associated with the control using htmlFor  jsx-a11y/label-has-for

/Users/justin/shakacode/react-on-rails/tmp/test-new-redux-generation/client/app/bundles/HelloWorld/containers/HelloWorld.jsx
   3:25  error  Absolute imports should come before relative imports  import/imports-first
   4:23  error  Absolute imports should come before relative imports  import/imports-first
  13:21  error  'updateName' is missing in props validation           react/prop-types

/Users/justin/shakacode/react-on-rails/tmp/test-new-redux-generation/client/app/bundles/HelloWorld/reducers/index.jsx
  8:42  error  Unexpected trailing comma  comma-dangle

/Users/justin/shakacode/react-on-rails/tmp/test-new-redux-generation/client/app/bundles/HelloWorld/reducers/person.jsx
  4:57  error  Unexpected trailing comma  comma-dangle

/Users/justin/shakacode/react-on-rails/tmp/test-new-redux-generation/client/app/bundles/HelloWorld/store/helloWorldStore.jsx
  3:23  error  Absolute imports should come before relative imports  import/imports-first
  9:29  error  Absolute imports should come before relative imports  import/imports-first

✖ 9 problems (9 errors, 0 warnings)


npm ERR! Darwin 15.6.0
npm ERR! argv "/Users/justin/.nvm/versions/node/v6.7.0/bin/node" "/Users/justin/.nvm/versions/node/v6.7.0/bin/npm" "run" "eslint"
npm ERR! node v6.7.0
npm ERR! npm  v3.10.3
npm ERR! code ELIFECYCLE
npm ERR! react-webpack-rails-tutorial@0.0.1 eslint: `eslint --ext .js,.jsx .`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the react-webpack-rails-tutorial@0.0.1 eslint script 'eslint --ext .js,.jsx .'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the react-webpack-rails-tutorial package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     eslint --ext .js,.jsx .
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs react-webpack-rails-tutorial
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls react-webpack-rails-tutorial
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/justin/shakacode/react-on-rails/tmp/test-new-redux-generation/client/npm-debug.log
```

File: client/app/bundles/HelloWorld/store/helloWorldStore.jsx (hunk 1 above)
Comment (mongjong59) [reply client/app/bundles/HelloWorld/store/helloWorldStore.jsx:9]: @justin808 Sure. I will remove thunk and Immutable.js and take care of linter. And I still need some clarification on this one: https://github.com/shakacode/react_on_rails-test-new-redux-generation/pull/1#discussion_r85885237

Comment (justin808) [reply client/app/bundles/HelloWorld/store/helloWorldStore.jsx:9]: Please use this linter setup: https://www.npmjs.com/package/eslint-config-shakacode

12.0.0-beta.1 

I just updated it to the latest.

Comment (justin808) [reply client/app/bundles/HelloWorld/store/helloWorldStore.jsx:9]: See https://github.com/shakacode/react_on_rails-test-new-redux-generation/pull/4

```
➜  ~/shakacode/react-on-rails/tmp/test-new-redux-generation/client (v2-with-linting) ✗ npm run eslint                                                                                        ✹ [19:41:10]

> react-webpack-rails-tutorial@0.0.1 eslint /Users/justin/shakacode/react-on-rails/tmp/test-new-redux-generation/client
> eslint --ext .js,.jsx .


/Users/justin/shakacode/react-on-rails/tmp/test-new-redux-generation/client/app/bundles/HelloWorld/actions/helloWorldActionCreators.jsx
  3:1  error  Prefer default export  import/prefer-default-export

/Users/justin/shakacode/react-on-rails/tmp/test-new-redux-generation/client/app/bundles/HelloWorld/components/HelloWorldWidget.jsx
  30:11  error  Form controls using a label to identify them must be programmatically associated with the control using htmlFor  jsx-a11y/label-has-for

/Users/justin/shakacode/react-on-rails/tmp/test-new-redux-generation/client/app/bundles/HelloWorld/containers/HelloWorld.jsx
   3:25  error  Absolute imports should come before relative imports  import/imports-first
   4:23  error  Absolute imports should come before relative imports  import/imports-first
  13:21  error  'updateName' is missing in props validation           react/prop-types

/Users/justin/shakacode/react-on-rails/tmp/test-new-redux-generation/client/app/bundles/HelloWorld/reducers/index.jsx
  8:42  error  Unexpected trailing comma  comma-dangle

/Users/justin/shakacode/react-on-rails/tmp/test-new-redux-generation/client/app/bundles/HelloWorld/reducers/person.jsx
  4:57  error  Unexpected trailing comma  comma-dangle

/Users/justin/shakacode/react-on-rails/tmp/test-new-redux-generation/client/app/bundles/HelloWorld/store/helloWorldStore.jsx
   3:23  error  Absolute imports should come before relative imports                                 import/imports-first
   9:29  error  Absolute imports should come before relative imports                                 import/imports-first
  11:24  error  Expected parentheses around arrow function argument having a body with curly braces  arrow-parens

✖ 10 problems (10 errors, 0 warnings)
```

**Pull request closed without merging**